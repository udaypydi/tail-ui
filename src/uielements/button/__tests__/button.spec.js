import React from 'react';
import { render } from 'test-utils';
import Button from 'uielements/button/button.component.js';

it('should take a snapshot', () => {
    const { asFragment } = render(<Button buttonText="test button" />);
  
    expect(asFragment(<Button buttonText="test button" />)).toMatchSnapshot();
});

it ('should render a button with text color', () => {
    const { asFragment } = render(<Button  color="green" />);
    
    expect(asFragment(<Button color="green"  />)).toMatchSnapshot();
});

it ('should render a button with custom styles', () => {
    const { asFragment } = render(<Button  style={{ background: '#0077b5', color: '#ffffff' }}  />);
    
    expect(asFragment(<Button  style={{ background: '#0077b5', color: '#ffffff' }}  />)).toMatchSnapshot();
});

it ('should render a button with custom tailwind classes', () => {
    const { asFragment } = render(<Button className="text-green-500 border-green-500" />);
    
    expect(asFragment(<Button className="text-green-500 border-green-500" />)).toMatchSnapshot();
});

it ('should render a rounded button', () => {
    const { asFragment } = render(<Button rounded />);
    
    expect(asFragment(<Button rounded />)).toMatchSnapshot();
});

it('should render a rounded button', () => {
    const { asFragment } = render(<Button rounded />);

    expect(asFragment(<Button rounded />)).toMatchSnapshot();
});
