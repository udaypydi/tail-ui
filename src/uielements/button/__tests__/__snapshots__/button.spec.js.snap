// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a button with custom styles 1`] = `
<DocumentFragment>
  <button
    class="
    bg-transparent-700 hover:bg-transparent-500 text-black-700 
    font-semibold hover:text-black-700 py-2 px-4 border border-black-700 "
    style="background: rgb(0, 119, 181); color: rgb(255, 255, 255);"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`should render a button with custom tailwind classes 1`] = `
<DocumentFragment>
  <button
    class="
    bg-transparent-700 hover:bg-transparent-500 text-black-700 
    font-semibold hover:text-black-700 py-2 px-4 border border-black-700 text-green-500 border-green-500"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`should render a button with text color 1`] = `
<DocumentFragment>
  <button
    class="
    bg-transparent-700 hover:bg-transparent-500 text-green-700 
    font-semibold hover:text-green-700 py-2 px-4 border border-black-700 "
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`should render a rounded button 1`] = `
<DocumentFragment>
  <button
    class="
    bg-transparent-700 hover:bg-transparent-500 text-black-700 
    font-semibold hover:text-black-700 py-2 px-4 border border-black-700 rounded-md "
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="
    bg-transparent-700 hover:bg-transparent-500 text-black-700 
    font-semibold hover:text-black-700 py-2 px-4 border border-black-700 "
    type="button"
  >
    test button
  </button>
</DocumentFragment>
`;
